src/widgets/modals/ByProgramModal.js

```diff
@@ -130,13 +142,19 @@ export const ByProgramModal = ({ settings, database, transactionType, onConfirm
     .map(({ description }) => description.toLowerCase());
 
   // Helper methods for fetching modal data for user selection
-  const getPrograms = () => getAllPrograms(settings, database);
+  const getPrograms = () => {
+    if (transactionType === 'customerRequisition') {
+      return getAllProgramsForCustomer(customer, database);
+    }
+    return getAllPrograms(settings, database);
```

Comment:
Finding the correct programs, given the customer (or no customer for supplier requisition)